### PR TITLE
Enable a disabled test

### PR DESCRIPTION
**Pull request by ant0nsc**
Undoing #520 

Please follow the guidelines for PRs contained [here](docs/pull_requests.md). Checklist:

- [x] Ensure that your PR is small, and implements one change.
- [x] Add unit tests for all functions that you introduced or modified.
- [x] Run PyCharm's code cleanup tools on your Python files.
- [x] Link the correct GitHub issue for tracking.
- [x] Update the [Changelog](CHANGELOG.md) file: Describe your change in terms of 
Added/Changed/Removed/... in the "Upcoming" section.
- [x] When merging your PR, replace the default merge message with a description of your PR,
and if needed a motivation why that change was required.

### CODE REVIEW

File: Tests/ML/visualizers/test_plot_cross_validation.py [hunk 1]
```diff
@@ -310,14 +312,22 @@ def test_result_aggregation_for_classification_all_epochs(test_output_dirs: Outp
                                                 expected_aggregate_metrics=expected_aggregates,
                                                 expected_epochs={1, 2, 3})
 
+
 @pytest.mark.after_training_ensemble_run
-def test_add_comparison_data(test_config_comparison: PlotCrossValidationConfig) -> None:
-    test_config_comparison.epoch = 2
-    metrics_df, root_folder = download_metrics(test_config_comparison)
```
Comment (JonathanTripp) [inline Tests/ML/visualizers/test_plot_cross_validation.py:316]: Isn't it better to modify the test_config_comparison test fixture? It's only used here. Or at least delete it

File: Tests/ML/reports/test_classification_report.py [hunk 2]
```diff
@@ -557,7 +576,8 @@ def test_get_image_outputs_from_subject_id(test_output_dirs: OutputFolderForTest
                                          LoggingColumns.ModelOutput.value: [0.1, 0.1, 0.1, 0.9, 0.1, 0.9,
                                                                             0.9, 0.9, 0.9, 0.9, 0.9, 0.1],
                                          LoggingColumns.Label.value: [0, 0, 0, 1, 0, 1, 1, 1, 1, 1, 1, 0],
-                                         LoggingColumns.CrossValidationSplitIndex: [DEFAULT_CROSS_VALIDATION_SPLIT_INDEX] * 12,
+                                         LoggingColumns.CrossValidationSplitIndex: [
```
Comment (JonathanTripp) [inline Tests/ML/reports/test_classification_report.py:579]: Why the newline?

File: Tests/ML/visualizers/test_plot_cross_validation.py (hunk 1 above)
Comment (ant0nsc) [reply Tests/ML/visualizers/test_plot_cross_validation.py:316]: good catch! deleted.

File: Tests/ML/reports/test_classification_report.py (hunk 2 above)
Comment (ant0nsc) [reply Tests/ML/reports/test_classification_report.py:579]: result of auto-formatting. cleaned up.

File: InnerEye/ML/deep_learning_config.py [hunk 3]
```diff
@@ -317,20 +285,34 @@ def inference_options(self) -> Dict[ModelProcessing, Dict[ModelExecutionMode, Op
...
+                    ModelExecutionMode.VAL: False,
+                    ModelExecutionMode.TEST: True,
+                }
+            },
+            ModelProcessing.ENSEMBLE_CREATION: {
+                True: {
+                    ModelExecutionMode.TRAIN: False,
+                    ModelExecutionMode.VAL: False,
+                    ModelExecutionMode.TEST: True,
+                }
```
Comment (JonathanTripp) [inline InnerEye/ML/deep_learning_config.py:309]: False and True branches are now the same, is this correct?

File: InnerEye/ML/reports/classification_crossval_report.ipynb [hunk 4]
```diff
@@ -199,8 +199,12 @@
    "outputs": [],
    "source": [
     "if test_metrics_csv.is_file():\n",
-    "    plot_pr_and_roc_curves_from_csv(metrics_csv=test_metrics_csv, data_split=ModelExecutionMode.TEST, config=config,\n",
-    "                                    is_crossval_report=is_crossval_report)"
+    "    # When analyzing cross validation runs, the report is started with a file that contains both\n",
+    "    # validation and test set metrics in one. The test set metrics however may be missing if\n",
+    "    # inference on the test set for cross validation child runs is turned off, leading to exceptions.\n",
+    "    if not is_crossval_report or (is_crossval_report and config.inference_on_test_set):\n",
+    "        plot_pr_and_roc_curves_from_csv(metrics_csv=test_metrics_csv, data_split=ModelExecutionMode.TEST, config=config,\n",
```
Comment (JonathanTripp) [inline InnerEye/ML/reports/classification_crossval_report.ipynb:206]: Is it cleaner to call is_inference_required here?

Comment (ant0nsc) [reply InnerEye/ML/reports/classification_crossval_report.ipynb:206]: no, I don't think so - entry to this code is already guarded sufficiently.

File: InnerEye/ML/deep_learning_config.py (hunk 3 above)
Comment (ant0nsc) [reply InnerEye/ML/deep_learning_config.py:309]: yes, I'd leave that in place though to make it more visible and easier to change if needed.

Comment (ant0nsc) [reply InnerEye/ML/deep_learning_config.py:309]: (yes, I have deliberately turned off validation set inference by default)

**Review by JonathanTripp — APPROVED** [review]

**Review by JonathanTripp — DISMISSED** [review]

**Review by dumbledad — APPROVED** [review]

**Review by JonathanTripp — APPROVED** [review]

**Pull request merged**